File: src/main/java/com/kigya/patterns/java/observer/subscribers/CurrentConditionsDisplay.java
package com.kigya.patterns.java.observer.subscribers;

import com.kigya.patterns.java.observer.publisher.WeatherData;
import lombok.extern.flogger.Flogger;
import org.jetbrains.annotations.NotNull;

import java.util.logging.Level;

@Flogger
public class CurrentConditionsDisplay implements Observer, DisplayElement {
    private float temperature;
    private float humidity;

    public CurrentConditionsDisplay(@NotNull WeatherData weatherData) {
        weatherData.registerObserver(this);
    }

    public void update(float temperature, float humidity, float pressure) {
        this.temperature = temperature;
        this.humidity = humidity;
        display();
    }

    public void display() {
        log.at(Level.INFO).log("Current conditions: " + temperature
                + "F degrees and " + humidity + "% humidity");
    }
}

File: src/main/java/com/kigya/patterns/kotlin/observer/subscibers/CurrentConditionsDisplay.kt
package com.kigya.patterns.kotlin.observer.subscibers

import com.kigya.patterns.kotlin.observer.publisher.Subject
import com.kigya.patterns.kotlin.observer.publisher.WeatherData
import com.kigya.patterns.kotlin.strategy.entity.logger

class CurrentConditionsDisplay(weatherData: Subject<WeatherData>) : Observer<WeatherData>, DisplayElement<WeatherData> {

    init {
        weatherData.registerObserver(this)
    }

    override fun update(data: WeatherData) = display(data)

    override fun display(data: WeatherData) =
        logger.info {
            "Current conditions: ${data.temperature} F degrees and ${data.humidity}% humidity"
        }
}